### ISSUE TITLE
CorsixTH 0.50 crash

### ISSUE THREAD

**Issue by mfukar** [CorsixTH/CorsixTH#923]
What steps will reproduce the problem?
1. Load the auto-saved game Autosave3.sav , found [here](https://www.dropbox.com/sh/oy30j9i3s4qj1cg/AAD2mRlzXK9pbA1l1mvupccXa?dl=0) (Dropbox link)
2. Let the game proceed
3. The game should crash with an Access violation - code c0000005

Version of CorsixTH as reported by the game window is 0.50, from the official release found [here](https://github.com/CorsixTH/CorsixTH/releases). OS is Windows 7, 64-bit.

The level played is full game level 4.

The gamelog is very short so I'm pasting it here:

```
Created game with savegame version 105.
Error in buttonup handler: 
E:\Programs\CorsixTH\Lua\entities\humanoid.lua:401: attempt to call a nil value (method 'isStanding')
stack traceback:
	E:\Programs\CorsixTH\Lua\entities\humanoid.lua:401: in method 'dump'
	E:\Programs\CorsixTH\Lua\entities\humanoid.lua:350: in field 'onClick'
	E:\Programs\CorsixTH\Lua\entities\patient.lua:76: in method 'onClick'
	E:\Programs\CorsixTH\Lua\ui.lua:627: in function <E:\Programs\CorsixTH\Lua\ui.lua:610>
	(...tail calls...)
	E:\Programs\CorsixTH\Lua\app.lua:779: in function <E:\Programs\CorsixTH\Lua\app.lua:774>
```

In case it's more helpful, here's a WinDbg stacktrace from the crash location:

```
0:000> kv
Child-SP          RetAddr           : Args to Child                                                           : Call Site
00000000`001af010 00000001`3fa55d81 : 00000000`00000106 00000000`0000004c 00000000`0000004b 00000001`3fa55cb0 : CorsixTH+0x43355
00000000`001af240 000007fe`f41bc2ea : 00000000`0e12c858 00000000`001af400 00000000`136da170 00000000`136d9f70 : CorsixTH+0x35d81
00000000`001af2b0 000007fe`f41d4aca : 00000000`136da170 00000000`00000010 00000000`00000008 00000000`001af400 : lua53!lua_setlocal+0x8ea
00000000`001af300 000007fe`f41bce1a : 00000000`01dba7b0 00000000`136da0f0 00000000`0de9fa50 00000000`00000001 : lua53!luaopen_utf8+0x1b7a
00000000`001af4b0 000007fe`f41bcbca : 00000000`0e12c858 00000000`00000001 00000000`00000001 00000000`00000001 : lua53!lua_yieldk+0x44a
00000000`001af4e0 000007fe`f41bc5e5 : 00000000`0e12c858 000007fe`ff882170 00000000`77b5991c 00000000`007b3100 : lua53!lua_yieldk+0x1fa
00000000`001af510 000007fe`f41bc933 : 00000000`0e12c858 00000000`0e12c858 00000000`001af730 00000001`3faf0e68 : lua53!lua_setlocal+0xbe5
00000000`001af690 00000001`3fa304b2 : 00000000`0e12c858 00000000`00000001 00000000`00000001 00000000`00000001 : lua53!lua_resume+0x53
00000000`001af6c0 000007fe`f41bc2ea : 00000000`0034d8c0 00000000`0032b788 00000000`0e1407a0 00000001`3fa2ff90 : CorsixTH+0x104b2
00000000`001af760 000007fe`f41d4aca : 00000000`0e1407a0 00000000`00000004 00000000`00000002 00000000`001af8b0 : lua53!lua_setlocal+0x8ea
00000000`001af7b0 000007fe`f41bbe14 : 00000000`00348ae0 00000000`0e140780 00000000`003a7410 00000000`00000001 : lua53!luaopen_utf8+0x1b7a
00000000`001af960 000007fe`f41b118b : 00000000`0032b788 00000000`00000020 00000000`00000001 00000000`00000000 : lua53!lua_setlocal+0x414
00000000`001af990 00000001`3fa71689 : 00000000`0032b788 00000000`00000000 00000001`3fa2284c 000007fe`ec7c7686 : lua53!lua_callk+0x5b
00000000`001af9c0 000007fe`f41bc2ea : 00000000`00000050 00000000`002d0000 00000000`00000042 00000000`00000042 : CorsixTH+0x51689
00000000`001afa00 000007fe`f41bbe08 : 00000000`00000004 00000000`0032b858 00000000`00000000 00000000`00000000 : lua53!lua_setlocal+0x8ea
00000000`001afa50 000007fe`f41bc5e5 : 00000000`0032b788 00000000`0032e420 00000000`0032b788 00000000`00000000 : lua53!lua_setlocal+0x408
00000000`001afa80 000007fe`f41bc030 : 00000000`5ea8d4b6 000007fe`f41cba39 00000000`00000000 000007fe`f41b5abe : lua53!lua_setlocal+0xbe5
00000000`001afc00 000007fe`f41b2444 : 00000000`00000001 00000000`00000000 00000000`00000000 00000000`0032b788 : lua53!lua_setlocal+0x630
00000000`001afc40 00000001`3fa710b2 : 00000000`00000000 00000000`001afd29 00000000`00329a10 00000000`00000000 : lua53!lua_pcallk+0xf4
00000000`001afc90 00000001`3fa25650 : 00000001`00000001 00000000`00329a10 00000000`00000000 00000000`00000000 : CorsixTH+0x510b2
00000000`001afd90 00000001`3fad00ab : 00000000`00000001 00000000`00329a10 00000000`00000000 00000000`00000000 : CorsixTH+0x5650
00000000`001afdd0 00000000`77a35a4d : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : CorsixTH+0xb00ab
00000000`001afe00 00000000`77c6b831 : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : kernel32!BaseThreadInitThunk+0xd
00000000`001afe30 00000000`00000000 : 00000000`00000000 00000000`00000000 00000000`00000000 00000000`00000000 : ntdll!RtlUserThreadStart+0x1d
```

If there's any more info I can provide, please let me know. Cheers, and keep up the great work!

**Comment by Alberth289346** [comment]
It doesn't crash here.

The stack trace suggests you clicked at something, did you?

**Comment by mfukar** [comment]
No I did not. I reproduced it just now without clicking anything at all after restoring the saved game.

**Comment by MarkL1961** [comment]
I don't get any errors either.

You said un-pause, but for me it is running fast, so have you shared the wrong save?

**Comment by mfukar** [comment]
Thanks for correcting me on that. The save is correct, but there's no need to unpause - will edit my initial comment.

**Comment by Alberth289346** [comment]
Upon further testing, Autosave does crash in 0.50, at Mar 11.
The reason is the window you have open on the patient walking back to home.

If you close it before the above date, the program will continue to run.
The bug itself has already been fixed in commit 7541804 already.
